### PR TITLE
designate: support for Openstack Application Credentials

### LINKED ISSUE #1218: Support Openstack Application Credentials
Openstack support [application credentials](https://docs.openstack.org/keystone/train/user/application_credentials.html) to create a "technical" user with less permissions. It would be nice if lego would support these application credentials. This would be partially solved by https://github.com/go-acme/lego/issues/1217 since you could specify the applications credentials in the `clouds.yml` file but we should also support setting the credentials over environment variables.

### PR DESCRIPTION
**Pull request by rissson**
Closes #1218 

This PR allows the use of `OS_APPLICATION_CREDENTIAL_*` to authenticate against designate.

As discussed in #1359, I included the update of the `github.com/gophercloud/gophercloud` and `github.com/gophercloud/utils` in this PR.

This PR removes some credentials checking that is already done by `gophercloud`, so the burden of credentials checking is not on `lego` anymore, but on `gophercloud`.

### CODE REVIEW

**Comment by rissson** [comment]
Just tested this with our production OpenStack (on victoria), it looks okay to me!

**Comment by ldez** [comment]
Could you add some the documentation about this feature in https://github.com/go-acme/lego/blob/master/providers/dns/designate/designate.toml
Then run `make generate-dns`.

You can use the `Additional` field to add the explanation
ex:
https://github.com/go-acme/lego/blob/2871fd3938788c6c63703028c56af63109bcce0f/providers/dns/cloudflare/cloudflare.toml#L18

**Comment by rissson** [comment]
Done! Thanks for your pointers :D

**Comment by ldez** [comment]
Could you also add some examples:
```toml
Example = '''
TODO
'''
```

**Review by ldez — APPROVED** [review]
Thanks :+1:

File: providers/dns/designate/designate.toml [hunk 1]
```diff
@@ -33,24 +33,16 @@ Additional = '''
...
+2. The second one is using your username and password, via the `OS_USERNAME`, `OS_PASSWORD` and `OS_PROJECT_NAME` environment variables.
+3. The third one is by using an application credential, via the `OS_APPLICATION_CREDENTIAL_*` and `OS_USER_ID` environment variables.
 
-The second one is using your username and password, via the `OS_USERNAME`,
-`OS_PASSWORD` and `OS_PROJECT_NAME` environment variables.
+For the username/password and application methods, the `OS_AUTH_URL` and `OS_REGION_NAME` environment variables are required.
 
-The third one is by using an application credential, via the
-`OS_APPLICATION_CREDENTIAL_*` and `OS_USER_ID` environment variables.
+For more information, you can read about the different method of authentication with OpenStack in then Keystone's documentation and then gophercloud documentation:
```
Comment (rissson) [inline providers/dns/designate/designate.toml:42]:
```
For more information, you can read about the different methods of authentication with OpenStack in the Keystone's documentation and the gophercloud documentation:
```

Comment (rissson) [reply providers/dns/designate/designate.toml:42]: Just some typo

**Review by rissson — COMMENTED** [review]
I forgot to re-generate the documentation, sorry about that.

**Comment by rissson** [comment]
Thanks for your very swift review!

**Pull request merged**